packages/lib/src/components/internal/FormFields/Select/Select.tsx

```diff
@@ -1,217 +1,201 @@
-import { Component, h } from 'preact';
+import { h } from 'preact';
+import { useEffect, useMemo, useRef, useState } from 'preact/hooks';
 import cx from 'classnames';
+import SelectButton from './components/SelectButton';
+import SelectList from './components/SelectList';
+import uuid from '../../../../utils/uuid';
+import { SelectItem, SelectProps } from './types';
 import styles from './Select.module.scss';
 import './Select.scss';
-import { SelectProps, SelectState } from './types';
 
-class Select extends Component<SelectProps, SelectState> {
-    private selectContainer;
-    private toggleButton;
-    private dropdownList;
-
-    public static defaultProps = {
-        items: [],
-        readonly: false,
-        onChange: () => {}
+function Select(props: SelectProps) {
+    const filterInputRef = useRef(null);
+    const selectContainerRef = useRef(null);
+    const toggleButtonRef = useRef(null);
+    const selectListRef = useRef(null);
+    const firstUpdate = useRef(true);
+    const [textFilter, setTextFilter] = useState<string>(null);
+    const [showList, setShowList] = useState<boolean>(false);
+    const selectListId: string = useMemo(() => `select-${uuid()}`, []);
+
+    const active: SelectItem = props.items.find(i => i.id === props.selected) || ({} as SelectItem);
+
+    const keys = {
+        arrowDown: 'ArrowDown',
+        arrowUp: 'ArrowUp',
+        enter: 'Enter',
+        escape: 'Escape',
+        space: ' '
     };
```

Comment:
This could also be declared outside of this component, and imported here.

Reply:
Done.